core/templates/pages/exploration-player-page/learner-experience/input-response-pair.component.ts

```diff
@@ -30,21 +30,22 @@ import { AudioTranslationManagerService } from '../services/audio-translation-ma
 import { PlayerPositionService } from '../services/player-position.service';
 import { PlayerTranscriptService } from '../services/player-transcript.service';
 import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
+import { Interaction } from 'domain/exploration/InteractionObjectFactory';
 
 @Component({
   selector: 'oppia-input-response-pair',
   templateUrl: './input-response-pair.component.html'
 })
 export class InputResponsePairComponent {
-  @Input() data: InputResponsePair;
+  @Input() data!: InputResponsePair;
   @Output() dataChange: EventEmitter<InputResponsePair> = new EventEmitter();
-  @Input() oppiaAvatarImageUrl: string;
-  @Input() profilePicture: string;
-  @Input() inputResponsePairId: string;
-  @Input() isLastPair: boolean;
-  OPPIA_AVATAR_LINK_URL: string;
-  decodedProfilePicture: string;
-  @ViewChild('popover') popover: NgbPopover;
+  @Input() oppiaAvatarImageUrl!: string;
+  @Input() profilePicture!: string;
+  @Input() inputResponsePairId!: string;
+  @Input() isLastPair!: boolean;
+  OPPIA_AVATAR_LINK_URL: string | undefined;
+  decodedProfilePicture: string | undefined;
```

Comment:
Why can these be `undefined`?

Reply:
Running strict checks throws an error, that is, "has no initializer and is not definitely assigned in the constructor" for these two variables. Is there any other way this can be handled without adding as 'undefined'?

Reply:
Is `OPPIA_AVATAR_LINK_URL` even used anywhere?

Reply:
Agreed, it wasn't used in this file. It's still used in a couple of files in other places though (but not from this class). I have removed it from this file after checking.